web-server/v0.4/components/Charts/Timeseries.js

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import { Chart, Tooltip, Geom, Legend, Axis } from 'bizcharts';
+import DataSet from '@antv/data-set';
+import Slider from 'bizcharts-plugin-slider';
+
+class Timeseries extends React.Component {
+  render() {
+    const {
+      title,
+      height = 400,
+      padding = [100, 100, 40, 100],
+      titleMap = {},
+      borderWidth = 2,
+      data = [],
+    } = this.props;
+
+    data.sort((a, b) => a.x - b.x);
+
+    var sortingQueue = []; 
+    var fields = [];
+    Object.keys(titleMap).map(iteration => {
+        fields.push(titleMap[iteration]);
+        sortingQueue.push([...data].sort((a, b) => b[iteration] - a[iteration])[0][iteration]);
+    })
+
+    let max;
+    if (data[0]) {
+      max = Math.max(...sortingQueue);
+    }
+
+    const ds = new DataSet({
+        state: {
+            start: data[0].x,
+            end: data[data.length - 1].x,
+        },
+    });
+
+    const dv = ds.createView();
+    dv
+      .source(data)
+      .transform({
```

Comment:
Can you add comments here which describe what the transforms are supposed to be doing?